### PR TITLE
Allow for precalculated sky samples only, speedup for sky marginalization

### PR DESCRIPTION
**Pull request by ahnitz**
This uses only draws from a precomputed set of sky points drawn from the SNRs of the reference signal. This avoids a l ot of computation in the inner loop of code for sky marginalization. In testing, some problems are sampler dominated in cost now. This is probably most appropriate for signals where the references is good and we expect only refinement from the basic bayestar-like sky localization. 

Depends on #4319

### CODE REVIEW

File: pycbc/inference/models/relbin.py
Comment (spxiwh) [inline pycbc/inference/models/relbin.py:535]: Is it faster this way? It seems that you still have the call to antenna_pattern ... Otherwise I don't understand the pol_phase change?

File: pycbc/inference/models/relbin.py
Comment (spxiwh) [inline pycbc/inference/models/relbin.py:537]: This copy definitely doesn't seem like it's that efficient compared to just using antenna_pattern with a nonzero pol_phase??

File: pycbc/inference/models/relbin.py
Comment (spxiwh) [inline pycbc/inference/models/relbin.py:687]: This doesn't look right. `snr_draw` should take snrs as input, not waveforms?!

File: pycbc/inference/models/relbin.py
Comment (spxiwh) [inline pycbc/inference/models/relbin.py:780]: This is what I think the code above should look like ...

File: pycbc/inference/models/relbin.py
Comment (spxiwh) [reply pycbc/inference/models/relbin.py:687]: Ah I see, you changed the function to use key-word arguments, but then didn't use it here because wfs comes first. Don't like that at all! If this is a key-word argument then give it as a key-word argument so that this is clear.

File: pycbc/inference/models/tools.py
Comment (spxiwh) [inline pycbc/inference/models/tools.py:316]: Do you need to check that wfs is not also None?

**Review by spxiwh — APPROVED** [review]
Minor changes requested, good to merge after they're done. I don't understand parts (most) of the changes in tools.py, so if you wanted to wait for Collin, he might be able to give a better review. But am happy that these changes are sufficiently contained that it doesn't break anything else, so it makes sense to merge now and change later if needed.

File: pycbc/inference/models/relbin.py
Comment (ahnitz) [reply pycbc/inference/models/relbin.py:687]: Good point, I've changed it to be explicit

**Pull request merged**
